// relatives-tree/lib/models/family.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Family {
    constructor(id, type, isMain = false) {
        this.id = id;
        this.type = type;
        this.main = isMain;
        this.pID = null;
        this.cID = null;
        this.top = 0;
        this.left = 0;
        this.pUnits = [];
        this.cUnits = [];
    }
    get width() {
        const units = this.cCount > this.pCount ? this.cUnits : this.pUnits;
        const lastUnit = units[units.length - 1];
        return lastUnit.shift + (2 * lastUnit.size);
    }
    get pCount() {
        return this.pUnits.reduce((a, b) => a + b.size, 0);
    }
    get cCount() {
        return this.cUnits.reduce((a, b) => a + b.size, 0);
    }
    get pUnitsWithParents() {
        return [...this.pUnits].filter(unit => (!!unit.nodes.find(node => !!node.parents.length)));
    }
    get cUnitsWithChildren() {
        return [...this.cUnits].filter(unit => (!!unit.nodes.find(node => !!node.children.length)));
    }
}
exports.default = Family;
